server/src/test/java/io/crate/integrationtests/DeleteIntegrationTest.java

```diff
@@ -303,4 +308,31 @@ public void test_delete_partitions_from_subquery_does_not_leave_empty_orphan_par
             "1"
         );
     }
+
+    @Test
+    public void test_can_reuse_prepared_statement_for_delete_containing_non_deterministic_function() throws Exception {
+        execute("CREATE TABLE doc.t (a timestamp with time zone)");
+
+        try (var session = sqlExecutor.newSession()) {
+            session.parse(
+                "preparedStatement",
+                "DELETE FROM doc.t WHERE a < now()",
+                List.of()
+            );
```

Comment:
Minor: can be make it `now() - '3 minute'::INTERVAL` to account for a more complex case that is covered now:
 top level function is deterministic but argument is not.

Reply:
That will make this test always pass

Reply:
I meant also adjusting INSERT. Basically whatever setup which would work but with non-deterministic argument